contrib/coredns/policy/policy.go

```diff
@@ -119,6 +119,52 @@ func (p *PolicyMiddleware) getEDNS0Attrs(r *dns.Msg) ([]*pb.Attribute, bool) {
 	return attrs, foundSourceIP
 }
 
+
+type LocalResponseWriter struct {
+	dns.ResponseWriter
+	Rcode int
+	Len int
+	Msg   *dns.Msg
+	localAddr  net.Addr
+	remoteAddr net.Addr
+}
+
+func New(w dns.ResponseWriter) *LocalResponseWriter {
+
+	var addr string = ""
+	l := &net.IPAddr{IP: net.ParseIP(addr)}
+	r := &net.IPAddr{IP: net.ParseIP(addr)}
+	return &LocalResponseWriter{
+		ResponseWriter: w,
+		Rcode:          0,
+		Msg:            nil,
+		localAddr: l,
+		remoteAddr: r,
+
+	}
+}
+
+func (r *LocalResponseWriter) WriteMsg(res *dns.Msg) error {
+	r.Rcode = res.Rcode
+	// We may get called multiple times (axfr for instance).
+	// Save the last message, but add the sizes.
+	r.Len += res.Len()
+	r.Msg = res
+	return r.ResponseWriter.WriteMsg(res)
```

Comment:
Actually, we do NOT want to do the WriteMsg here. The point is we want to just capture the `res` in r.Msg so we can do the WriteMsg *later* if we decide to keep it.